examples/module/src/gizmo/__init__.py

```diff
@@ -10,3 +11,4 @@
 Registry.register_subtype(ResourceRegistration(Gizmo, GizmoService, lambda name, channel: GizmoClient(name, channel)))
 
 Registry.register_resource_creator(Gizmo.SUBTYPE, MyGizmo.MODEL, MyGizmo.new)
+Registry.register_resource_validator(Gizmo.SUBTYPE, MyGizmo.MODEL, MyGizmo.validate_config)
```

Comment:
I would actually roll this into the resource creator registry using a dataclass (so we'd register the creator and validator at the same time), like how we do it in Golang and C++